src/libs/polycube/include/polycube/services/cube_iface.h

```diff
@@ -103,6 +103,11 @@ class TransparentCubeIface : virtual public BaseCubeIface {
 
   virtual void set_conf(const nlohmann::json &conf) = 0;
   virtual nlohmann::json to_json() const = 0;
+
+  virtual void subscribe_event(const std::string &param_name,
+          std::function<void(const std::string, const std::string)> callback) = 0;
```

Comment:
`std::function<void(const std::string, const std::string)> callback` is used in many places, maybe it is a good idea to define a new type for this.